pkg/services/accesscontrol/middleware.go

```diff
@@ -39,6 +41,13 @@ func authorize(c *models.ReqContext, ac AccessControl, user *user.SignedInUser,
 	}
 
 	hasAccess, err := ac.Evaluate(c.Req.Context(), user, injected)
+
+	// If the request contains the access control test header we short circuit successful checks and returns 200
+	if c.Req.Header.Get(AccessControlTestHeader) == "1" && hasAccess {
```

Comment:
Shouldn't this be disabled on production environment? 
Hypothetically, could a malicious user use this header to quickly enumerate the api endpoints to look for misconfigurations?

What do you think?